### ISSUE TITLE
Site config cleanup

### ISSUE THREAD

**Issue by AlexanderRichert-NOAA** [JCSDA/spack-stack#1116]
For the 1.8.0 release, let's do the following for each site config:
 - Verify OSes
 - Remove old/unused compilers
 - Check compiler versions
 - Check external package paths/versions
 - Create a test environment, concretize, and install a few packages

Site list:
 - [x] acorn (@AlexanderRichert-NOAA)
 - [x] atlantis (@climbfuji)
-  [x] blueback (@climbfuji)
 - [x] ~~aws-pcluster~~ currently not supported
 - [x] casper - move to tier2 (@climbfuji)
 - [x] derecho
 - [x] ~~discover-scu16~~ no longer supported
 - [x] discover-scu17
 - [x] ~~emc-rhel~~ currently not supported
 - [x] ~~frontera~~ no longer supported
 - [x] gaea-c5
 - [x] gaea-c6
 - [x] hera
 - [x] hercules
 - [x] ~~jet~~ no longer supported
 - [x] linux.default
 - [x] macos.default
 - [x] narwhal (@climbfuji)
 - [x] nautilus (@climbfuji)
 - [x] noaa-aws
 - [x] noaa-azure
 - [x] noaa-gcloud
 - [x] orion
 - [x] s4

**Comment by climbfuji** [comment]
@AlexanderRichert-NOAA I did quite a bit of a cleanup in my last PR (the MKL PR #759). Given that, and the fact that we have to revisit all the sites when we remove the "revert to openblas" logic, are you ok with pushing this out to after the 1.8.0 release?

**Comment by AlexanderRichert-NOAA** [comment]
Yep that's fine. Plus the automation work will be a good opportunity to review site configs.

**Comment by climbfuji** [comment]
This is all covered by the site config updates for spack-stack-1.9.1. Will close as completed once 1.9.1 is available everywhere and the release code is tagged with all site config changes included.
